content/ember/v5/deprecate-array-prototype-extensions.md

```diff
@@ -0,0 +1,1265 @@
+  @action
+  insertAt(idx, value) {
+    this.abc.splice(idx, 0, value);
+  }
+}
+```
+or
+
+```js
+// @tracked
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+
+export default class SampleComponent extends Component {
+  @tracked abc = ['x', 'y', 'z', 'x'];
+
+  @action
+  insertAt(idx, value) {
+    this.abc = [...this.abc.slice(0, idx), value, this.abc.slice(this.abc.length - idx)]
+  }
+}
+```
+
+#### `popObject`
+Before
+```js
+import Component from '@glimmer/component';
+export default class SampleComponent extends Component {
+  abc = ['x', 'y', 'z', 'x'];
+
+  @action
+  popObject() {
+    this.abc.popObject();
+  }
+}
+```
+
+After
+```js
+// TrackedArray
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { TrackedArray } from 'tracked-built-ins';
+
+export default class SampleComponent extends Component {
+  abc = new TrackedArray(['x', 'y', 'z', 'x']);
+
+  @action
+  popObject() {
+    this.abc.pop();
+  }
+}
+```
+or
+
+```js
+// @tracked
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+
+export default class SampleComponent extends Component {
+  @tracked abc = ['x', 'y', 'z', 'x'];
+
+  @action
+  popObject() {
+    this.abc.pop();
+    this.abc = [...this.abc];
+  }
+}
+```
+
+#### `pushObject`
+Before
+```js
+import Component from '@glimmer/component';
+export default class SampleComponent extends Component {
+  abc = ['x', 'y', 'z', 'x'];
+
+  @action
+  pushObject(value) {
+    this.abc.pushObject(value);
+  }
+}
+```
+
+After
+```js
+// TrackedArray
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { TrackedArray } from 'tracked-built-ins';
+
+export default class SampleComponent extends Component {
+  abc = new TrackedArray(['x', 'y', 'z', 'x']);
+
+  @action
+  pushObject(value) {
+    this.abc.push(value);
+  }
+}
+```
+or
+
+```js
+// @tracked
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+
+export default class SampleComponent extends Component {
+  @tracked abc = ['x', 'y', 'z', 'x'];
+
+  @action
+  pushObject(value) {
+    this.abc = [...this.abc, value];
+  }
+}
+```
+
+#### `pushObjects`
+Before
+```js
+import Component from '@glimmer/component';
+export default class SampleComponent extends Component {
+  abc = ['x', 'y', 'z', 'x'];
+
+  @action
+  pushObjects(values) {
+    this.abc.pushObjects(values);
+  }
+}
+```
+
+After
+```js
+// TrackedArray
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { TrackedArray } from 'tracked-built-ins';
+
+export default class SampleComponent extends Component {
+  abc = new TrackedArray(['x', 'y', 'z', 'x']);
+
+  @action
+  pushObjects(values) {
+    this.abc.splice(this.abc.length, 0, ...values);
```

Comment:
Would `this.abc.push(...values)` work here? It would be simpler and match the first `after` example for `pushObject` above?

Reply:
yeah, i think so!